src/core/api/ip6_api.cpp

```diff
@@ -135,7 +135,7 @@ void otIp6SetReceiveFilterEnabled(otInstance *aInstance, bool aEnabled)
 
 otError otIp6Send(otInstance *aInstance, otMessage *aMessage)
 {
-    return AsCoreType(aInstance).Get<Ip6::Ip6>().SendRaw(AsCoreType(aMessage));
+    return AsCoreType(aInstance).Get<Ip6::Ip6>().SendRaw(AsCoreType(aMessage), /* aFromNcpHost */ false);
```

Comment:
How about adding a config here to determine the value of param `aFromNcpHost` (something like `OPENTHREAD_CONFIG_IP6_ALLOW_PASSING_USER_MSG_IN_CALLBACK` or perhaps a better name) and use it here instead.

This way if this change causes issue for some users/developers, we can ask them to change this config (otherwise, they may need to update the source code directly).

Reply:
Added configuration `OPENTHREAD_CONFIG_IP6_ALLOW_LOOP_BACK_HOST_DATAGRAMS`.